static/js/public.js

```diff
@@ -121,3 +121,37 @@ jQuery(document).ready(function ($) {
     emailBody: description + CURRENT_PAGE_URL
   });
 });
+
+/**
+ * Set url hash if hash provided in the url,
+ * or set hash based on the active panel
+ */
+$(function(){
+  $('.mdl-tabs__tab').click(function(){
+    document.location.hash = $(this).attr('href');
+  });
+  if (document.location.hash){
+    setPanelActive(document.location.hash);
+  } else {
+    location.hash = $('.mdl-tabs__tab.is-active').attr('href');
+  }
+});
+
+$(window).on('hashchange', function () {
+  if (location.hash) {
+    setPanelActive(location.hash);
+  }
+});
+
+/**
+ *  Given a valid hash, set the corresponding panel active.
+ */
+function setPanelActive(hash){
+  if ($(hash).length > 0) {
+    $(".mdl-tabs__panel, .mdl-tabs__tab").removeClass('is-active');
+    $(hash).addClass('is-active');
```

Comment:
Nitpick: you could create a variable at the start of this function and reference that here, instead of calling `$(hash)` multiple times.
```js
var $hash = $(hash);
if ($hash.length > 0) {
  $hash.addClass('is-active');
  // ...
```